homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -408,17 +560,47 @@ async def async_setup_entry(
         entities.extend(
             EnvoyProductionPhaseEntity(coordinator, description)
             for use_phase, phase in envoy_data.system_production_phases.items()
-            for description in PRODUCTION_PHASE_SENSORS[PhaseNames(use_phase)]
+            for description in PRODUCTION_PHASE_SENSORS[use_phase]
             if phase is not None
         )
     # For each consumption phase reported add consumption entities
     if envoy_data.system_consumption_phases:
         entities.extend(
             EnvoyConsumptionPhaseEntity(coordinator, description)
             for use_phase, phase in envoy_data.system_consumption_phases.items()
-            for description in CONSUMPTION_PHASE_SENSORS[PhaseNames(use_phase)]
+            for description in CONSUMPTION_PHASE_SENSORS[use_phase]
             if phase is not None
         )
+    # Add net consumption CT entities
+    if ctmeter := envoy_data.ctmeter_consumption:
+        entities.extend(
+            EnvoyConsumptionCTEntity(coordinator, description)
+            for description in CT_NET_CONSUMPTION_SENSORS
+            if ctmeter.measurement_type == CtType.NET_CONSUMPTION
+        )
+    # For each net consumption ct phase reported add net consumption entities
+    if phase_data := envoy_data.ctmeter_consumption_phases:
+        entities.extend(
+            EnvoyConsumptionCTPhaseEntity(coordinator, description)
+            for use_phase, phase in phase_data.items()
+            for description in CT_NET_CONSUMPTION_PHASE_SENSORS[use_phase]
+            if phase.measurement_type == CtType.NET_CONSUMPTION
+        )
+    # Add production CT entities
+    if ctmeter := envoy_data.ctmeter_production:
+        entities.extend(
+            EnvoyProductionCTEntity(coordinator, description)
+            for description in CT_PRODUCTION_SENSORS
+            if ctmeter.measurement_type == CtType.PRODUCTION
```

Comment:
```suggestion
            if ctmeter.measurement_type is CtType.PRODUCTION
```

Please use identity check for enums

Reply:
Oh `measurement_type` isn't actually a `CtType` its a `str` so it not an enum compare

Reply:
Ah yes, comes straight from the Envoy. json